lib/src/markdown_processor.dart

```diff
@@ -775,17 +784,146 @@ class Documentation {
       // Assume the user intended Dart if there are no other classes present.
       if (!specifiesLanguage) pre.classes.add('language-dart');
     }
+    String asHtml;
+    String asOneLiner;
 
-    // `trim` fixes issue with line ending differences between mac and windows.
-    var asHtml = asHtmlDocument.body.innerHtml?.trim();
-    var asOneLiner = asHtmlDocument.body.children.isEmpty
+    if (processFullDocs) {
+      // `trim` fixes issue with line ending differences between mac and windows.
+      asHtml = asHtmlDocument.body.innerHtml?.trim();
+    }
+    asOneLiner = asHtmlDocument.body.children.isEmpty
         ? ''
         : asHtmlDocument.body.children.first.innerHtml;
-    if (!asOneLiner.startsWith('<p>')) {
-      asOneLiner = '<p>$asOneLiner</p>';
+
+    return new Tuple2(asHtml, asOneLiner);
+  }
+
+  // From package:markdown/src/document.dart
+  // TODO(jcollins-g): consider making this a public method in markdown package
+  void _parseInlineContent(List<md.Node> nodes) {
+    for (int i = 0; i < nodes.length; i++) {
+      var node = nodes[i];
+      if (node is md.UnparsedContent) {
+        List<md.Node> inlineNodes =
+            new md.InlineParser(node.textContent, this).parse();
+        nodes.removeAt(i);
+        nodes.insertAll(i, inlineNodes);
+        i += inlineNodes.length - 1;
+      } else if (node is md.Element && node.children != null) {
+        _parseInlineContent(node.children);
+      }
+    }
+  }
+
+  /// Returns a tuple of longHtml, shortHtml (longHtml is NULL if !processFullDocs)
+  Tuple3<String, String, bool> renderLinesToHtml(
+      List<String> lines, bool processFullDocs) {
+    bool hasExtendedDocs = false;
+    md.Node firstNode;
+    List<md.Node> nodes = [];
+    for (md.Node node
+        in new IterableBlockParser(lines, this).parseLinesGenerator()) {
+      if (firstNode != null) {
+        hasExtendedDocs = true;
+        if (!processFullDocs) break;
+      }
+      if (firstNode == null) {
+        firstNode = node;
+      }
+      nodes.add(node);
+    }
+    _parseInlineContent(nodes);
+
+    String shortHtml;
+    String longHtml;
+    if (processFullDocs) {
+      Tuple2 htmls = _renderNodesToHtml(nodes, processFullDocs);
+      longHtml = htmls.item1;
+      shortHtml = htmls.item2;
+    } else {
+      if (firstNode != null) {
+        Tuple2 htmls = _renderNodesToHtml([firstNode], processFullDocs);
+        shortHtml = htmls.item2;
+      } else {
+        shortHtml = '';
+      }
+    }
+    if (!shortHtml.startsWith('<p>')) {
```

Comment:
To be honest, I have no idea -- I left it in because the old code did that and some of the tests seemed to verify it.  However, I can see nothing that actually requires this -- docs look and browse OK even without these tags.  Removing.  This will trigger a large set of meaningless test package doc changes and test changes.